src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int NOT_EXISTS = -1;
+    private static final int UNIT = 1;
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] internalArray;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        internalArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= capacity) {
+            growIfArrayFull();
+        }
+        internalArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == index) {
```

Comment:
This method could be simplified I suggest to create one method that will check if index is ok instead of 'size == index' and !isIndexAcceptable(index)